realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -671,6 +671,12 @@ enum StringPredicate {
 void TableQuery_StringPredicate(JNIEnv *env, jlong nativeQueryPtr, jlongArray columnIndexes, jstring value, jboolean caseSensitive, StringPredicate predicate) {
     GET_ARRAY()
     try {
+        if (value == NULL) {
+            if (!(Q(nativeQueryPtr)->get_table()->is_nullable( S(arr[0]) ))) {
+                ThrowException(env, IllegalArgument, "Trying to query field with null but field is not nullable.");
+                return;
+            }
+        }
         bool isCaseSensitive = caseSensitive ? true : false;
```

Comment:
What is the style guide for variables. So far I thought we used ```camelCase``` for java variabels and ```under_score_case``` for c++ variabels?